### PR TITLE
Pass transaction GUID from client for ProcessInventoryResetQuantity

### PR DESCRIPTION
**Pull request by bospoort**
## Description

Updated the `ProcessInventoryResetQuantity` stored procedure and `AdjustQuantityModal` component to implement client-side transaction GUID generation for idempotency. This prevents duplicate transaction submissions when users retry failed requests and ensures consistency with other inventory operations (`ProcessInventoryChange` and checkout processes).

Key improvements:
- Transaction GUIDs are now generated on the client using `crypto.randomUUID()`
- Duplicate transaction detection prevents the same transaction from being processed twice
- Improved error handling with proper validation before transaction starts
- Comprehensive unit tests ensure all edge cases are covered

## Jira Ticket
- Closes: [PIT-400](https://das-ph-inventory-tracker.atlassian.net/browse/PIT-400)

## Type of Change
**Type:** New feature / Bug fix

## Changes Made
- **Database Stored Procedure** (`process_inventory_reset_quantity.sql`):
  - Added `@new_transaction_id UNIQUEIDENTIFIER` parameter
  - Implemented duplicate transaction check before starting transaction
  - Returns `DUPLICATE_TRANSACTION` error code for duplicate submissions
  - Improved error handling with `XACT_ABORT` for automatic rollback
  - All validations now happen before transaction starts to prevent partial commits

- **React Component** (`AdjustQuantityModal.tsx`):
  - Added transaction ID state management with `useState`
  - Generates new UUID when dialog opens using `crypto.randomUUID()`
  - Added validation to ensure transaction ID exists before submission
  - Updated API call to pass `new_transaction_id` parameter
  - Implemented `DUPLICATE_TRANSACTION` error handling with auto-close after 2 seconds
  - Improved error handling: validates response.ok, checks for valid JSON, handles empty responses

- **Unit Tests** (`test_process_inventory_reset_quantity.sql` - new file):
  - Test successful quantity reset
  - Test duplicate transaction ID detection
  - Test multiple sequential resets with different transaction IDs
  - Test invalid item ID handling
  - Test negative quantity validation
  - Test reset to zero (edge case)
  - Test transaction type verification (CORRECTION = 3)
  - Automatic cleanup/restore after tests

- **Updated existing test** (`test_inventory_change.sql`):
  - Added test cases demonstrating ProcessInventoryResetQuantity with transaction GUID

## Checklist
- [x] My code follows the project's style guidelines
- [x] I have run [prettier](https://github.com/digitalaidseattle/plymouth-housing#code-formatting) on the code
- [x] I have performed a self-review of my code
- [x] I have commented my code only in hard-to-understand areas
- [x] I have updated the documentation accordingly
- [x] My changes generate no new warnings in Chrome Dev Tools
- [x] I have added unit tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### Testing Steps:
1. **Successful Quantity Reset:**
   - Navigate to inventory page
   - Click "Adjust Quantity" on any item
   - Enter a new quantity value
   - Submit the form
   - Verify: Item quantity is updated, success message appears

2. **Duplicate Transaction Prevention:**
   - Open browser dev tools Network tab
   - Adjust an item's quantity
   - In the Network tab, find the POST request to `PROCESS_INVENTORY_RESET_QUANTITY`
   - Right-click and "Replay" the request
   - Verify: Error message appears: "This transaction has already been submitted."
   - Verify: Modal auto-closes after 2 seconds
   - Verify: Item quantity remains unchanged (not double-applied)

3. **Error Handling:**
   - Try setting a negative quantity (should show validation error before submission)
   - Try with invalid item (server error should be handled gracefully)

4. **SQL Unit Tests:**
   - Run the test file: `database/tests/test_process_inventory_reset_quantity.sql`
   - Verify all tests pass with "PASS" messages in output

### Expected Behavior:
- Transaction IDs are unique per form submission
- Duplicate submissions are rejected with clear error message
- No partial transactions or orphaned transaction records
- Consistent with existing inventory operations (Add Inventory, Checkout)

## Summary by CodeRabbit

* **New Features**
  * Inventory adjustments now include a required unique transaction ID and include it in submissions.
  * Added "How do you know this?" and Comments fields to the adjustment form.

* **Improvements**
  * Validation enforces non-negative whole-number quantities and checks transaction ID before processing.
  * Clearer server error handling and specific messaging for duplicate or invalid inputs; improved success feedback.

* **Tests**
  * Comprehensive automated tests added to validate success, duplicates, invalid item/quantity, and logging.

### CODE REVIEW

File: database/procedures/process_inventory_reset_quantity.sql [hunk 1]
```diff
@@ -5,68 +5,93 @@ CREATE PROCEDURE ProcessInventoryResetQuantity
...
     @item_id INT,
     @new_quantity INT,
-    @additional_notes NVARCHAR(MAX)
+    @additional_notes NVARCHAR(MAX),
+    @new_transaction_id UNIQUEIDENTIFIER
 AS
 BEGIN
     SET NOCOUNT ON;
     SET XACT_ABORT ON;
+    -- Auto-rollback, never explicit ROLLBACK
```
[the remark this answers is not in the document]
Comment (bospoort) [reply database/procedures/process_inventory_reset_quantity.sql]: RaiseError with error level above 10 will roll back. Relying on rollback will drop all error information. DAB swallows it.

[the remark this answers is not in the document]
Comment (bospoort) [reply database/procedures/process_inventory_reset_quantity.sql]: when the error comes out of the EXEC calls to logtransaction, ROLLBACK fails

File: src/components/inventory/AdjustQuantityModal.tsx [hunk 2]
```diff
@@ -18,186 +30,310 @@ type AdjustQuantityModalProps = {
...
   const [errorMessage, setErrorMessage] = useState('');
   const [isSubmitting, setIsSubmitting] = useState(false);
+  const [transactionId, setTransactionId] = useState<string | null>(null);
+
+  // Generate a new transaction ID when the dialog opens
+  useEffect(() => {
+    if (showDialog) {
+      setTransactionId(crypto.randomUUID());
+    }
+  }, [showDialog]);
```
[the remark this answers is not in the document]
Comment (bospoort) [reply src/components/inventory/AdjustQuantityModal.tsx]: We always use HTTPS and we have control over the browser base. Not an issue
